chain/syncer.go

```diff
@@ -324,14 +290,34 @@ func (syncer *Syncer) widen(ctx context.Context, ts types.TipSet) (types.TipSet,
 	return wts, nil
 }
 
+// NewPeerHeadInfo accepts a peer, its current head state and its trustworthiness
+// and returns a PeerHeadInfo.
+func NewPeerHeadInfo(head types.TipSetKey, height types.Uint64, from peer.ID, trusted bool) *PeerHeadInfo {
+	return &PeerHeadInfo{
+		Head:    head,
+		Height:  height,
+		From:    from,
+		Trusted: trusted,
+	}
+}
+
+// PeerHeadInfo contains information about a peers current head state and
+// its trustworthiness.
+type PeerHeadInfo struct {
+	Head    types.TipSetKey
+	Height  types.Uint64
+	From    peer.ID
+	Trusted bool
+}
+
 // HandleNewTipset extends the Syncer's chain store with the given tipset if they
 // represent a valid extension. It limits the length of new chains it will
 // attempt to validate and caches invalid blocks it has encountered to
 // help prevent DOS.
-func (syncer *Syncer) HandleNewTipset(ctx context.Context, tsKey types.TipSetKey, from peer.ID) (err error) {
-	logSyncer.Debugf("Begin fetch and sync of chain with head %v", tsKey)
+func (syncer *Syncer) HandleNewTipset(ctx context.Context, ph *PeerHeadInfo) (err error) {
```

Comment:
While you're here, consider renaming to `HandleNewTipSet` to match project capitalization convention.